### PR TITLE
Extend setup_packages.py to bing package with its dependencies

### PR DESCRIPTION
**Pull request by JanuszL**
- in some cases, the package should be installed with another one
  in well-defined version. For example, tensorflow-gpu 2.6 should be
  used only with tensorflow-estimator==2.6.0, while other TF versions
  could use any
- this PR adds an ability to add to each package relevant dependencies

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

## Description
- [ ] **Bug fix** (*non-breaking change which fixes an issue*)
- [ ] **New feature** (*non-breaking change which adds functionality*)
- [ ] **Breaking change** (*fix or feature that would cause existing functionality to not work as expected*)
- [ ] **Refactoring** (*Redesign of existing code that doesn't affect functionality*)
- [x] **Other** (*e.g. Documentation, Tests, Configuration*)

### What happened in this PR
- in some cases, the package should be installed with another one
  in well-defined version. For example, tensorflow-gpu 2.6 should be
  used only with tensorflow-estimator==2.6.0, while other TF versions
  could use any
- this PR adds an ability to add to each package relevant dependencies

#### Additional information
- Affected modules and functionalities:
  - setup_packages.py

- Key points relevant for the review:
  - if the approach is correct

### Tests
- [x] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: qa/setup_packages.py [hunk 1]
```diff
@@ -92,6 +100,21 @@ def get_alias(version):
         """
         return getattr(version, "alias", None)
 
+
+    def get_dependencies(self, cuda_version=None, idx=None):
+        """Obtains dependant packages list if exists. Otherwise return None
```
Comment (awolant) [inline qa/setup_packages.py:105]: This doc made me check, what happens if you pass `None` to `join`. Unsurprisingly, it fails :)

**Review by awolant — APPROVED** [review]

File: qa/setup_packages.py [hunk 2]
```diff
@@ -28,12 +28,16 @@ class PckgVer():
             Maximum python version supported by this package. If empty there is no upper bound
         `python_min_ver`: str, optional, default = None
             Mimimum python version supported by this package. If empty there is no lower bound
+        `dependant_packages` : list of str, optiona, default = None
```
Comment (jantonguirao) [inline qa/setup_packages.py:31]:
```suggestion
        `dependencies` : list of str, optional, default = None
```
or
```suggestion
        `dependent_packages` : list of str, optional, default = None
```

File: qa/setup_packages.py (hunk 1 above)
Comment (jantonguirao) [reply qa/setup_packages.py:105]: I believe the doc is not accurate now. We are returning empty string, not None

File: qa/setup_packages.py [hunk 3]
```diff
@@ -183,7 +206,9 @@ def get_install_string(self, idx, cuda_version=None):
             `cuda_version`: str, optional, default = None
                 Cuda version used for this query
         """
-        return "{name}=={version}".format(name=self.get_name(cuda_version, idx), version=self.get_version(idx, cuda_version))
+        pkg_cmd = "{name}=={version}".format(name=self.get_name(cuda_version, idx), version=self.get_version(idx, cuda_version))
+        deps_cmd = " ".join(self.get_dependencies(cuda_version, idx))
+        return " ".join([pkg_cmd, deps_cmd])
```
Comment (jantonguirao) [inline qa/setup_packages.py:211]:
```suggestion
        return " ".join([pkg_cmd] + self.get_dependencies(cuda_version, idx))
```
would there be any difference with this? I mean, can we make it a single `join` call?

File: qa/setup_packages.py (hunk 2 above)
Comment (JanuszL) [reply qa/setup_packages.py]: Fixed

File: qa/setup_packages.py (hunk 1 above)
Comment (JanuszL) [reply qa/setup_packages.py]: Fixed

File: qa/setup_packages.py (hunk 3 above)
Comment (JanuszL) [reply qa/setup_packages.py]: Fixed

File: qa/setup_packages.py (hunk 1 above)
Comment (JanuszL) [reply qa/setup_packages.py]: Fixed

**Review by jantonguirao — APPROVED** [review]

**Review by awolant — APPROVED** [review]

**Review by jantonguirao — APPROVED** [review]

**Pull request merged**